src/allmydata/util/eliotutil.py

```diff
@@ -324,8 +324,8 @@ def decorate_log_call_deferred(f):
         def logged_f(*a, **kw):
             # Use the action's context method to avoid ending the action when
             # the `with` block ends.
-            args = {k: bytes_to_unicode(True, kw[k]) for k in kw}
-            with start_action(action_type=action_type, **args).context():
+            kwargs = {k: bytes_to_unicode(True, kw[k]) for k in kw}
+            with start_action(action_type=action_type, args=a, **kwargs).context():
```

Comment:
I think it would be better to log keyword arguments as `kwargs=kwargs` rather than `**kwargs`.  This avoids a collision between the contents of `kwargs` and the keyword arguments being passed statically.

Consider:

```
>>> def f(a, b): pass
... 
>>> f(a=1, **{"b": 2})
>>> f(a=1, **{"a": 2})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: f() got multiple values for keyword argument 'a'
>>> 
```

So, specifically, if `log_call_deferred` is ever used on a function that accepts `action_type` or `args` as a keyword argument then it will fail.

Reply:
Right.

Reply:
@meejah Please can you take a look one more time?

Reply:
That had already been implemented, infact `kwargs` where the only arguments logged by default. The whole point of having `args` and `kwargs` passed to `start_action` is to prevent argument collision. All arguments must exist within on of those two namespaces, in this case what is the point of the test for collision? Or are we talking about collision within the namespaces themselves (that raise an error).